src/components/frame/useFrame.ts

```diff
@@ -43,26 +53,42 @@ interface UseChildrenFrameProps {
 export const useChildFrame = function (
   props: UseChildrenFrameProps
 ): [boolean, { dispatch?: HostActionsHandler } & Partial<LegacyHostActions>] {
-  const [parentFrameConnection, setParentFrameConnection] = useState<any>();
+  const [childFrameConnection, setChildFrameConnection] = useState<any>();
   const [status, setStatus] = useState<Status>("DISCONNECTED");
 
   useEffect(() => {
-    const run = async (): Promise<void> => {
-      setParentFrameConnection(
-        await connectToChild({
-          methods: {
-            dispatch: props.dispatch,
-            ...props.methods,
-          },
-          iframe: props.iframe.current,
-        }).promise
-      );
-      setStatus("CONNECTED");
-    };
     if (props.iframe.current && status === "DISCONNECTED") {
+      const childV5 = connectToChild({
+        methods: {
+          dispatch: props.dispatch,
+          ...props.methods,
+        },
+        iframe: props.iframe.current,
+        timeout: 5000, // this will ensure connection to promise reject, when connection versions not match. otherwise it will be stucked in promise pending
```

Comment:
I'm scared that 5s timeout is a bit low, for instance cold start of server + react :grimacing:

Reply:
ok 9000 then

Reply:
why 9 000 ?

Reply:
no particular reason, if 5 digits the finger need to apply extra keystroke

Reply:
Lol let's go with 30s instead (same as browser default timeout for some stuff)

Reply:
ok no prob